File: java/src/main/java/mx/iteso/Priviledge/Delete.java
package mx.iteso.Priviledge;

import mx.iteso.PriviledgeDecorator;
import mx.iteso.User;

public class Delete extends PriviledgeDecorator {
    User user;

    public Delete (User user){
        this.user = user;
    }

    @Override
    public String getDescription() {
        return user.getDescription() + " can delete";
    }
}

File: csharp/decorator-practice/decorator-practice/Delete.cs
namespace decorator_practice
{
    internal class Delete : PrivilegesDecorator
    {
        private readonly User _user;


        public Delete(User userIn)
        {
            _user = userIn;
        }

        override
        public string PublishM()
        {
            return _user.PublishM();
        }

        override
        public string DeleteM()
        {
            return "Can delete";
        }

        override
        public string ApproveM()
        {
            return _user.ApproveM();
        }

        override
        public string CommentM()
        {
            return _user.CommentM();
        }
    }
}
